### ISSUE TITLE
SNS subscription filter applied too late after subscription creation

### ISSUE THREAD

**Issue by carlos-zaragoza** [terraform-providers/terraform-provider-aws#10495]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

Terraform v0.12.10

### Affected Resource(s)

* aws_sns_topic_subscription

### Terraform Configuration Files

```hcl
terraform {
  backend "local" {}
  required_version = "~> 0.12.10"
}

provider "aws" {
    region = "us-east-2"
}

# premade topic so debug output is more focused
data "aws_sns_topic" "foo" {
  name = "foo"
}

# premade queue so debug output is more focused
data "aws_sqs_queue" "bar" {
  name = "bar"
}

resource "aws_sns_topic_subscription" "baz" {
  topic_arn = data.aws_sns_topic.foo.arn
  protocol  = "sqs"
  endpoint  = data.aws_sqs_queue.bar.arn
  raw_message_delivery = "true"

  filter_policy = <<EOF
{
  "mykey": [
    "value_one",
    "value_two"
  ]
}
EOF
}
```

### Debug Output

https://gist.github.com/carlos-zaragoza/c0de7eb00582eb19fb0e450e2d4aaa8e

### Panic Output

N/A

### Expected Behavior

Subscription was created with its `FilterPolicy` applied simultaneously at time of creation.

### Actual Behavior

Subscription was created and then the `FilterPolicy` was applied afterwards.

In the example shown in the gist above, the `FilterPolicy` was only applied about 1 second later. When dealing with high volume SNS Topics though (something I have dealt with), 1 second of an unfiltered subscription can bring in dozens of unwanted messages or more. Those messages may not be handled correctly by an application not designed for them.

The AWS CLI is capable of creating both at the same time.

### Steps to Reproduce

1. `terraform apply`

### Important Factoids

N/A

**Comment by carlos-zaragoza** [comment]
Just saw the instruction that says to leave a comment if you've submitted a PR for this issue. So here's the link https://github.com/terraform-providers/terraform-provider-aws/pull/10496.

**Comment by bflad** [comment]
This change has been merged and will release with version 3.10.0 of the Terraform AWS Provider, late next week. Thanks to @carlos-zaragoza for the implementation. 👍
